### ISSUE TITLE
Post processing fails if both nzbToModia and Sickrage are set to move

### ISSUE THREAD

**Issue by Jelmerb** [clinton-hall/nzbToMedia#1160]
Hi Clinton Hall

Suddenly post processing doesn't work anymore. I don't get any errors but the files just disappear. It used to work just fine and I didn't change anything. I pinpointed the problem to the situation where I set both the processing method for Sickrage and the useLink in the autoProcessMedia.cfg to move. In the logs nothing unusual happens, but the file is never moved to the destination. When I change useLink to hard in autoProcessMedia.cfg post processing works, but I want Transmission to stop seeding the file (using move). Why isn't Sickrage post processing working when both Sickrage and useLink are set to move? 

Kind regards,

Jelmer

**Comment by Jelmerb** [comment]
2017-01-07 23:20:36 INFO POSTPROCESSOR-MANUAL :: Manual post processing task for /volume1/************/complete/tv/This.is.Us.S01E01.HDTV.x264-KILLERS[ettv] completed
2017-01-07 23:20:36 INFO POSTPROCESSOR-MANUAL :: Successfully processed
2017-01-07 23:20:36 INFO POSTPROCESSOR-MANUAL :: Processing succeeded for /volume1/************/complete/tv/This.is.Us.S01E01.HDTV.x264-KILLERS[ettv]/This.is.Us.S01E01.HDTV.x264-KILLERS[ettv].mkv
2017-01-07 23:20:34 INFO POSTPROCESSOR-MANUAL :: Using nzb_name for release name.
2017-01-07 23:20:34 INFO POSTPROCESSOR-MANUAL :: This download is marked a priority download so I'm going to replace an existing file if I find one
2017-01-07 23:20:34 INFO POSTPROCESSOR-MANUAL :: Processing /volume1/************/complete/tv/This.is.Us.S01E01.HDTV.x264-KILLERS[ettv]/This.is.Us.S01E01.HDTV.x264-KILLERS[ettv].mkv (This.is.Us.S01E01.HDTV.x264-KILLERS[ettv])
2017-01-07 23:20:33 INFO POSTPROCESSOR-MANUAL :: PostProcessing Path: /volume1/************/complete/tv/This.is.Us.S01E01.HDTV.x264-KILLERS[ettv]
2017-01-07 23:20:33 INFO POSTPROCESSOR-MANUAL :: Beginning manual post processing task: /volume1/************/complete/tv/This.is.Us.S01E01.HDTV.x264-KILLERS[ettv]
2017-01-07 23:20:32 INFO Thread-14 :: Manual post processing task for /volume1/************/complete/tv/This.is.Us.S01E01.HDTV.x264-KILLERS[ettv] was added to the queue
2017-01-07 23:20:32 INFO TORNADO :: User logged into the SickRage web interface
2017-01-07 23:20:32 INFO TORNADO :: User logged into the SickRage web interface

this is the log I get from Sickrage when Processing method for Sickrage is set to move and useLink in autoProcessMedia.cfg is set to hard under [Torrent]. Then the file is actually processed.

But when I set both Sickrage processing method and useLink to move I get these messages in the log:
2017-01-07 23:56:03 INFO POSTPROCESSOR-MANUAL :: Manual post processing task for /volume1/************/complete/tv/This.Is.Us.S01E03.HDTV.x264-KILLERS[ettv] completed
2017-01-07 23:56:03 INFO POSTPROCESSOR-MANUAL :: Beginning manual post processing task: /volume1/************/complete/tv/This.Is.Us.S01E03.HDTV.x264-KILLERS[ettv]
2017-01-07 23:56:00 INFO Thread-15 :: Manual post processing task for /volume1/************/complete/tv/This.Is.Us.S01E03.HDTV.x264-KILLERS[ettv] was added to the queue
2017-01-07 23:55:46 INFO TORNADO :: User logged into the SickRage web interface
2017-01-07 23:55:33 INFO TORNADO :: User logged into the SickRage web interface

Does it have something to do with the new queue Sickrage added that was mentioned in a previous post? 

I tried setting delete_original to 1, useLink to hard and processing method in Sickrage to move, so that the torrent is not seeding after download, but this also breaks post processing.

**Comment by clinton-hall** [comment]
Can you show the nzbToMedia log for the above process event?
If I had to guess... with Hard link, the file exists in original download location and OutputDirectory. With move, it only exists in OutPutDirectory and now (following recent queue implementation) SickRage is now checking the download complete location?

**Comment by Jelmerb** [comment]
2017-01-08 14:07:49 INFO    ::MAIN: Loading config from [/volume1/transmission/nzbToMedia/autoProcessMedia.cfg]
2017-01-08 14:07:49 INFO    ::MAIN: Checking database structure...
2017-01-08 14:07:49 INFO    ::MAIN: Checking if git needs an update
2017-01-08 14:07:50 INFO    ::MAIN: No update needed
2017-01-08 14:07:50 INFO    ::MAIN: nzbToMedia Version:66abb2d562aa9491c5c5fad0c228e20a05982fd7 Branch:master (Linux 3.10.77)
2017-01-08 14:07:51 WARNING ::MAIN: Failed to locate ffmpeg. Transcoding disabled!
2017-01-08 14:07:51 WARNING ::MAIN: Install ffmpeg with x264 support to enable this feature  ...
2017-01-08 14:07:51 WARNING ::MAIN: Failed to locate ffprobe. Video corruption detection disabled!
2017-01-08 14:07:51 WARNING ::MAIN: Install ffmpeg with x264 support to enable this feature  ...
2017-01-08 14:07:51 INFO    ::MAIN: #########################################################
2017-01-08 14:07:51 INFO    ::MAIN: ## ..::[TorrentToMedia.py]::.. ##
2017-01-08 14:07:51 INFO    ::MAIN: #########################################################
2017-01-08 14:07:51 INFO    ::MAIN: SEARCH: Found torrent directory This.is.Us.S01E02.HDTV.x264-KILLERS[ettv] in input directory directory /volume1/transmission/complete/tv
2017-01-08 14:07:51 INFO    ::MAIN: SEARCH: Setting inputDirectory to /volume1/transmission/complete/tv/This.is.Us.S01E02.HDTV.x264-KILLERS[ettv]
2017-01-08 14:07:51 INFO    ::MAIN: Auto-detected SECTION:SickBeard
2017-01-08 14:07:56 INFO    ::MAIN: Output directory set to: /volume1/transmission/complete/tv/This.is.Us.S01E02.HDTV.x264-KILLERS[ettv]
2017-01-08 14:07:56 INFO    ::COPYLINK: MEDIAFILE: [This.is.Us.S01E02.HDTV.x264-KILLERS[ettv].mkv]
2017-01-08 14:07:56 INFO    ::COPYLINK: SOURCE FOLDER: [/volume1/transmission/complete/tv/This.is.Us.S01E02.HDTV.x264-KILLERS[ettv]]
2017-01-08 14:07:56 INFO    ::COPYLINK: TARGET FOLDER: [/volume1/transmission/complete/tv/This.is.Us.S01E02.HDTV.x264-KILLERS[ettv]]
2017-01-08 14:07:56 INFO    ::COPYLINK: SOURCE AND TARGET files are the same, skipping ...
2017-01-08 14:07:56 INFO    ::MAIN: FLATTEN: Flattening directory: /volume1/transmission/complete/tv/This.is.Us.S01E02.HDTV.x264-KILLERS[ettv]
2017-01-08 14:07:56 INFO    ::MAIN: Found 1 media files in /volume1/transmission/complete/tv/This.is.Us.S01E02.HDTV.x264-KILLERS[ettv]
2017-01-08 14:07:56 INFO    ::MAIN: Calling SickBeard:tv to post-process:This.is.Us.S01E02.HDTV.x264-KILLERS[ettv]
2017-01-08 14:07:56 INFO    ::MAIN: Attempting to auto-detect tv fork
2017-01-08 14:07:57 INFO    ::MAIN: SickBeard:tv fork auto-detection successful ...
2017-01-08 14:07:57 INFO    ::MAIN: SickBeard:tv fork set to sickrage
2017-01-08 14:07:57 POSTPROCESS::SICKBEARD: SUCCESS: The download succeeded, sending a post-process request
2017-01-08 14:07:57 POSTPROCESS::SICKBEARD: Manual post processing task for /volume1/transmission/complete/tv/This.is.Us.S01E02.HDTV.x264-KILLERS[ettv] was added to the queue
2017-01-08 14:07:57 POSTPROCESS::SICKBEARD: <br\><span class='hidden'>Processing succeeded</span>
2017-01-08 14:08:02 INFO    ::CLEANDIR: Directory /volume1/transmission/complete/tv/This.is.Us.S01E02.HDTV.x264-KILLERS[ettv] has been processed and removed ...
2017-01-08 14:08:02 INFO    ::MAIN: The /volume1/transmission/nzbToMedia/TorrentToMedia.py script completed successfully.

this is the nzbToMedia log when post processing fails.

**Comment by Jelmerb** [comment]
What is your recommended setup with Sickrage when I want to move the file to the destination folder and remove the torrent from Transmission?

**Comment by Jelmerb** [comment]
Hi Clinton, I fix it by changing outputDirectory to a different directory then the inputDirectory where Transmission saves the completed downloads. Now post processing works with both useLink and Sickrage set to move. I don't know why it stopped working but at least it is working again. 
When I have useLink set to hard and Sickrage processing method to move, I sometimes get this error from Sickrage:

2017-01-08 15:41:06 POSTPROCESSOR-AUTO :: Processing failed for /volume1/video/temp/tv/This.Is.Us.S01E09.WEB-DL.XviD-FUM[ettv]/This.Is.Us.S01E09.WEB-DL.XviD-FUM[ettv].avi: File is locked for reading/writing
2017-01-08 15:41:06 POSTPROCESSOR-AUTO :: Problem(s) during processing, failed the following files/folders: 
2017-01-08 15:41:06 POSTPROCESSOR-AUTO :: [/volume1/video/temp/tv/This.Is.Us.S01E09.WEB-DL.XviD-FUM[ettv]/This.Is.Us.S01E09.WEB-DL.XviD-FUM[ettv].avi : Processing failed: File is locked for reading/writing]

Maybe it is a bug in Sickrage, maybe in nzbToMedia, but I thought you would like to know.

**Comment by Jelmerb** [comment]
Why does nzbToMedia move the file to outputDirectory and then Sickrage moves it to another location? It seems to me like an unnecessary action.

**Comment by clinton-hall** [comment]
Ok, I see what is going on.

First to answer the question. "Why does nzbToMedia move the file to outputDirectory and then Sickrage moves it to another location?"

Because nzbToMedia can do some renaming, transcoding, etc before handing to SickRage, and for people who want to continue seeding, this is the safest way to avoid any issues. For hard-ling (or move) on the same files system, this is near instantaneous.

the whole change here is with SickRage, By now adding the "queue" function, nzbToMedia now things SickRage has finished, and with option move, nzbToMedis goes ahead and deletes the torrent (so in same directory the files disappear BEFORE SickRage renames). With the separate directories as you have set, the issue is that the Torrent resumes seeding (applies the files lock) or deletes (during cleanup) before SickRage finishes renaming...

In short. I need to change the say this fork of SickRage is handled in nzbToMedia because the current return from SickRage doesn't confirm when the files have been renamed.

**Comment by Jelmerb** [comment]
Thank you for extensive answer. :) I understand it. Although as I mentioned in my second last post, I fixed the problem with setting a different outputDirectory than the inputDirectory. There is no issue now strangely. The locked file issue is resolved because the Transmission stops seeding when the file gets moved. But the renaming issue should with Sickrage shoud still be there. Maybe the moving operation in nzbToMedia takes just enough time for Sickrage to rename the file?? Anyways thanks your help.
